src/Microsoft.HttpRepl.Telemetry/UserLevelCacheWriter.cs

```diff
@@ -48,6 +48,8 @@ public sealed class UserLevelCacheWriter : IUserLevelCacheWriter
 
         public string RunWithCache(string cacheKey, Func<string> getValueToCache)
         {
+            getValueToCache = getValueToCache ?? throw new ArgumentNullException(nameof(getValueToCache));
```

Comment:
fun fact: `_ = getValueToCache ?? throw new ArgumentNullException(nameof(getValueToCache));` will save you 2 ops in MSIL, IIRC.

Reply:
I like it.